iphone/Classes/TiUISearchBar.m

```diff
@@ -54,6 +54,19 @@ -(UISearchBar*)searchBar
 	return searchView;
 }	
 
+-(void)setSearchBar:(UISearchBar *)searchBar
+{
+    if (searchView)
+    {
+        [searchView removeFromSuperview];
+        [searchView setDelegate:nil];
+        RELEASE_TO_NIL(searchView);
+    }
+    searchView = searchBar;
+    [searchView setDelegate:self];
+    [self addSubview:searchView];
+}
```

Comment:
First it is removing the search bar then adding it as a subview. So not a problem.